core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -536,6 +538,17 @@ trait EntityComponent {
 
         val sourceShardId = determineShard(clonedWorkspaceId)
         val destShardId = determineShard(newWorkspaceId)
+
+        var entityCount = 0
+        workspaceQuery.findByIdOrFail(clonedWorkspaceId.toString).map(w => {
+          listEntities(w).map(e => {
+            entityCount += 1
+            e
+          })
+          clonedWorkspaceEntityHistogram += entityCount
+          clonedWorkspaceAttributeHistogram += w.attributes.size
+        })
+
```

Comment:
check out `cloneEntitiesToNewWorkspace` at line 917 - I think it already returns/calculates all of the counts you need, without involving an extra database query!